### PR TITLE
Refactor Mina RPCs (compatible)

### PR DESCRIPTION
**Pull request by nholland94**
This PR refactors the Mina RPC implementations such that they are now in a single unified location and have more shared/less duplicated logic.

The first commit is the largest and most annoying to review, but it mostly moves type definitions around and combines RPC implementations that were split across `Mina_networking` and `Mina_lib`. It's intended to have limited behavior changes (the changes that were made are just in standardizing how RPCs are handled). Subsequent commits make more meaningful changes to RPCs, including refactoring the node status system out of the RPC layer (since that was moved over to libp2p_helper some time ago). Further changes will be included in a PR building off of this into develop.

### CODE REVIEW

**Comment by nholland94** [comment]
!ci-build-me

**Comment by nholland94** [comment]
!ci-build-me

**Comment by nholland94** [comment]
!ci-build-me

File: src/lib/mina_lib/tests/tests.ml [hunk 1]
```diff
@@ -459,18 +411,35 @@ let%test_module "Epoch ledger sync tests" =
...
+          Linear_pipe.create ()
+        in
+        don't_wait_for (
+          Linear_pipe.Reader.iter proxy_answer_reader ~f:(fun answer ->
+            if Option.is_none response && is_genesis_state_hash
+            ( match response with
+              | None when genesis_state_hash ->
+            )
+            Linear_pipe.write proxy_response_writer response)
+        *)
```
Comment (mrmr1993) [inline src/lib/mina_lib/tests/tests.ml:429]: Delete?

**Review by mrmr1993 — APPROVED** [review]

**Comment by nholland94** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**